src/System.Management.Automation/engine/hostifaces/LocalConnection.cs

```diff
@@ -863,12 +863,24 @@ private void DoCloseHelper()
                 }
 
                 if ((hostRunspace == null) || (this == hostRunspace))
-                {
+                {                   
+                    // We should close transcripting only if we are closing the last opened runspace.
+                    foreach (Runspace runspace in RunspaceList)
+                    {
+                        // At this stage, the last opened runspace should be at closing state.
+                        if (runspace.RunspaceStateInfo.State == RunspaceState.Opened)
+                        {
+                            return;
+                        }
+                    }
+
                     PSHostUserInterface host = executionContext.EngineHostInterface.UI;
                     if (host != null)
                     {
                         host.StopAllTranscribing();
                     }
+
+                    AmsiUtils.Uninitialize();
```

Comment:
AmsiUtils.Uninitialize() is still being called at line 930 below.  Please remove that call so that uninitialize only happens when the last runspace closes.